asteroid/data/medleydb_dataset.py

```diff
@@ -0,0 +1,243 @@
+import torch
+from torch.utils import data
+import json
+import os
+import numpy as np
+import soundfile as sf
+import random
+import torchaudio
+
+
+class MedleydbDataset(data.Dataset):
+    """MedleyDB: a dataset of annotated, royaltyfree multitrack recordings.
+
+    It provides a stereo mix and both dry and processed multitrack stems for 
+    each song in the dataset. The dataset covers a wide distribution of genres 
+    and primarily consists of full length songs with professional 
+    or nearprofessional audio quality.
+    
+    The dataset consists of 254 full lengths music tracks of
+    different genres. It provides a stereo mix and both dry and processed
+    multitrack stems for each song. 
+
+    This dataset contains a .yml object for each multitrack containining
+    metadata for each audio file in the multitrack. This dataloader utilises 
+    this metadata to select audio files and downmix them to create mixtures
+    of instruments belonging to a list of instrument tags with specified 
+    polyphony.
+
+
+    Args:
+        json_dir (str): Path containing .json file with list of audio files
+        n_src (int): Number of separate instrument classes (yet to be implemented)
+        n_poly (int): Number of instances of each class to be present in mixture.
```

Comment:
So `n_poly` will be the number of track mixed together to form the mixture right? 
And `n_src` is the total number of instruments usable in the dataset?

Reply:
(If my understanding is correct)
All the other dataset call `n_src`, the number of sources per mixture. 
I think it'll be confusing that in this case `n_poly` is the number of sources. 
Can we change `n_poly` to `n_src` and `n_src` to `n_instr` or something please?

Reply:
That's a good suggestion. I'll make that change. As of now the number of instruments supported is only 1 (since we considering only one class i.e. vocals). I will subsequently add the functionality to contain multiple instrument groups.

The structure is desired to be as follows. A list of tags for a given instrument type. For eg for a mixture of 2 vocals and 1 guitar:
Instrument 1: {male vocals, female vocals, choir, male rapper, female rapper}
Instrument 2: {electric guitar, distorted guitar, acoustic guitar}

n_inst: 2
n_src: [2 1]

As of now since support is only for vocal mixtures, n_inst will always be 1, and n_src should be int only.